adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java

```diff
@@ -60,4 +61,23 @@ public Route addRoute(final Route newRoute) throws SQLIntegrityConstraintViolati
             return newRoute;
         }
     }
+
+    private final Map<String, FareAttribute> fareAttributeCollection = new HashMap<>();
+
+    @Override
+    public FareAttribute getFareAttributeByFareId(final String fareId) {
+        return fareAttributeCollection.get(fareId);
+    }
+
+    @Override
+    public FareAttribute addFareAttribute(final FareAttribute newFareAttribute)
```

Comment:
While I appreciate you trying to find the right exception to throw here, I don't think `java.sql.SQLIntegrityConstraintViolationException ` is the right one. We're not using SQL here. 

If we stick to using exceptions for this type of data validation (which, IMHO, we shouldn't - see my other comment), then these types of errors should probably be our own subclass of `Exception`, something like `GtfsUniqueIdException`.

Reply:
Finally, since we are not throwing Exceptions, this method returns null, then a null check if executed in the use case execution method, see: 
https://github.com/MobilityData/gtfs-validator/blob/9f6f3a0d99bbdc2f08ed9bce682efdbacdcd84c7/usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedFareAttribute.java#L75